packages/sdk/src/utils/EventSubscription.ts

```diff
@@ -0,0 +1,166 @@
+import { AnimationPlaybackType, FrameAnimationType } from '../types/AnimationTypes';
+import { BarcodeFrameValidationResult } from '../types/BarcodeTypes';
+import { CharacterStyle } from '../types/CharacterStyleTypes';
+import { DocumentColor } from '../types/ColorStyleTypes';
+import { ConfigType, RuntimeConfigType, SelectedLayoutFrame, Id, AsyncError } from '../types/CommonTypes';
+import { AuthRefreshRequest, AuthCredentials, ConnectorEvent, ConnectorInstance } from '../types/ConnectorTypes';
+import { UndoState } from '../types/DocumentTypes';
+import { DocumentFontFamily } from '../types/FontTypes';
+import { FrameLayoutType, Frame } from '../types/FrameTypes';
+import { LayoutPropertiesType, MeasurementUnit, LayoutListItemType } from '../types/LayoutTypes';
+import { PageSize } from '../types/PageTypes';
+import { ParagraphStyle } from '../types/ParagraphStyleTypes';
+import { CornerRadiusUpdateModel } from '../types/ShapeTypes';
+import { SelectedTextStyle } from '../types/TextStyleTypes';
+import { Variable } from '../types/VariableTypes';
+import { ViewMode } from '../types/ViewModeTypes';
+import { Viewport } from '../types/ViewportTypes';
+import { ToolType } from './Enums';
+
+class EventSubscriptionBase {}
+
+export class EventSubscription<T extends (...args: any[]) => any> extends EventSubscriptionBase {
+    private callbacks: Map<string, T> = new Map();
+    legacyEventHandler: (() => T | undefined) | undefined;
+
+    constructor(legacyEventHandler?: () => T | undefined) {
+        super();
+        this.legacyEventHandler = legacyEventHandler;
+    }
+
+    // Register a new callback with an optional name
+    registerCallback(callback: T, name?: string): string {
+        const callbackName = name || this.generateUniqueName();
+        this.callbacks.set(callbackName, callback);
+        return callbackName;
+    }
+
+    // Unsubscribe a callback by its name
+    unsubscribeCallback(name: string): void {
+        this.callbacks.delete(name);
+    }
+
+    // Execute all registered callbacks with given arguments
+    executeCallbacks(...args: Parameters<T>): any {
+        let result;
+
+        if (this.legacyEventHandler) {
+            const callback = this.legacyEventHandler();
+            if (callback) {
+                result = callback(...args);
+            }
+        }
+
+        for (const callback of this.callbacks.values()) {
+            result = callback(...args);
+        }
+
+        return result;
+    }
+
+    // Generate a unique name for unnamed callbacks
+    private generateUniqueName(): string {
+        return `callback_${Math.random().toString(36).substr(2, 9)}`;
+    }
+}
+
+export class EventHelper {
+    static ensureSubscriptions(config: ConfigType): RuntimeConfigType {
+        // we need a reference to the clone object in the eventsubscriptions, so we need to create the object in 2 steps
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
```

Comment:
a bit dangerous to use it, might bring some unpredictable mistakes

Reply:
This was the best I could do, I know you TS skills are superior, feel free to take a shot at it. I added tests that cover all the essential parts, so if they stay green, you will be good